java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java

```diff
@@ -1278,6 +1279,60 @@ void testCast() {
     }
   }
 
+  @Test
+  void testCastTimestampAsString() {
+    final String[] TIMES_S_STRING = {
+        "2018-07-04 12:00:00",
+        "2023-01-25 07:32:12",
+        "2018-07-04 12:00:00"};
+
+    final long[] TIMES_S = {
+        1530705600L,   //'2018-07-04 12:00:00'
+        1674631932L,   //'2023-01-25 07:32:12'
+        1530705600L};  //'2018-07-04 12:00:00'
+
+    final String[] UNSUPPORTED_TIME_S_STRING = {"1965-10-26 14:01:12",
+        "1960-02-06 19:22:11"};
+
+    final long[] UNSUPPORTED_TIME_S = {-131968728L,   //'1965-10-26 14:01:12'
+        -312439069L};   //'1960-02-06 19:22:11'
+
+    final long[] TIMES_NS = {
+        1530705600115254330L,   //'2018-07-04 12:00:00.115254330'
+        1674631932929861604L,   //'2023-01-25 07:32:12.929861604'
+        1530705600115254330L};  //'2018-07-04 12:00:00.115254330'
+
+    final long[] UNSUPPORTED_TIME_NS = {-131968727761702469L};   //'1965-10-26 14:01:12.238297531'
+
+    final String[] TIMES_NS_STRING = {
+        "2018-07-04 12:00:00.115254330",
+        "2023-01-25 07:32:12.929861604",
+        "2018-07-04 12:00:00.115254330"};
+
+    final String[] UNSUPPORTED_TIME_NS_STRING = {"1965-10-26 14:01:12.238297531"};
+
+    // Seconds
+    try (ColumnVector s_string_times = ColumnVector.fromStrings(TIMES_S_STRING);
+         ColumnVector s_timestamps = ColumnVector.timestampSecondsFromLongs(TIMES_S);
+         ColumnVector unsupported_s_string_times = ColumnVector.fromStrings(UNSUPPORTED_TIME_S_STRING);
+         ColumnVector unsupported_s_timestamps = ColumnVector.timestampSecondsFromLongs(UNSUPPORTED_TIME_S);
+         ColumnVector timestampsAsStrings = s_timestamps.asStrings("%Y-%m-%d %H:%M:%S")) {
+      assertColumnsAreEqual(s_string_times, timestampsAsStrings);
+      assertThrows(AssertionFailedError.class, () -> assertColumnsAreEqual(unsupported_s_string_times, unsupported_s_timestamps));
```

Comment:
This is a bug in cudf. The rationale behind this is test will fail when it gets fixed.

Reply:
Does that make sense? Should I change it to a disabled test?

Reply:
Tests should not verify that broken behavior is broken.  Why do we want the build to fail when a bug is fixed? If the bug is a blocker for release then you put in the test and let the build fail until it is fixed.  If it is not a blocker for release you put in the test and disable it so you can still release, but you have the test ready to go so you can verify that the bug is fixed when it is fixed and that it does not regress in the future.

Split this large test up into several smaller tests so we have ones that pass with correct values, and we have some that would fail with the incorrect values, but those tests are disabled and are pointing to the issue that is intended to fix them.